lib/phlex/table.rb

```diff
@@ -0,0 +1,97 @@
+module Phlex
+  class Table < Phlex::Component
+    def initialize(resources, layout: :column)
+      raise ArgumentError unless [:column, :row].include?(layout)
+
+      @resources = resources
+      @layout = layout
+      @properties = {}.compare_by_identity
+    end
+
+    def template(&block)
+      table do
+        original_size = @_target.size
+        yield(self)
+        unchanged = (original_size == @_target.size)
+
+        if @properties.any?
+          raise ArgumentError unless unchanged
+
+          case @layout
+          when :column
+            column_layout
+          when :row
+            row_layout
+          end
+        end
+      end
+    end
+
+    def property(header, &block)
+      if header.is_a?(String)
+        header_text = header
+        header = -> { text header_text }
+      end
+
+      @properties[header] = block
+    end
+
+    def head(*args, **kwargs, &block)
+      thead(*args, **kwargs, &block)
+    end
+
+    def body(*args, **kwargs, &block)
+      tbody(*args, **kwargs, &block)
+    end
+
+    def header(*args, **kwargs, &block)
+      th(*args, **kwargs, &block)
+    end
+
+    def row(*args, **kwargs, &block)
+      tr(*args, **kwargs, &block)
+    end
+
+    def cell(*args, **kwargs, &block)
+      td(*args, **kwargs, &block)
+    end
```

Comment:
The default implementations for these could be faster alias methods if we changed the way we define tag methods.